spec/lib/params_sanitizer_spec.rb

```diff
@@ -0,0 +1,160 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+describe "ParamsSanitizer", type: :concern do
+  describe "instance methods" do
+    let(:params) { { "xml" => nil } }
+    subject { ParamsSanitizer.new(params) }
+
+    context "responds to its methods" do
+      it { expect(subject).to respond_to(:cleanse) }
+    end
+  end
+
+
+  describe "add_xml" do
+    context "when xml is nil" do
+      let(:params) { { "xml" => nil } }
+      subject { ParamsSanitizer.new(params) }
+
+      it "should return nil" do
+        expect(subject.add_xml).to eq(nil)
+      end
+    end
+
+    context "when xml is empty" do
+      let(:params) { { "xml" => "" } }
+      subject { ParamsSanitizer.new(params) }
+      it "should return nil" do
+        expect(subject.add_xml).to eq(nil)
+      end
+    end
+
+    context "when xml is not string" do
+      let(:params) { { "xml" => 123 } }
+      subject { ParamsSanitizer.new(params) }
+      it "should return nil" do
+        expect(subject.add_xml).to eq(nil)
+      end
+    end
+
+    context "when xml is string" do
+      let(:params) { { "xml" => "<xml></xml>" } }
+      subject { ParamsSanitizer.new(params) }
+      it "should return hash" do
+        expect(subject.add_xml).to eq(nil)
+      end
+    end
+  end
+
+  describe "add_metadata_version" do
+    context "when params is nil" do
+      let(:params) { { "version_info" => nil } }
+      subject { ParamsSanitizer.new(params) }
+
+      it "should return nil" do
+        subject.add_metadata_version(params)
+        expect(subject.get_params[:version_info]).to eq(nil)
+      end
+    end
+
+    context "when params is empty" do
+      let(:params) { { "version_info" => "" } }
+      subject { ParamsSanitizer.new(params) }
+      it "should return nil" do
+        subject.add_metadata_version(params)
+        expect(subject.get_params[:version_info]).to eq(nil)
+      end
+    end
+
+    context "when params is not hash" do
+      let(:params) { { "version_info" => 123 } }
+      subject { ParamsSanitizer.new(params) }
+      it "should return nil" do
```

Comment:
This description does not match the assertion, I think it should be "should return correct value" or similar